### PR TITLE
feat: Add bucketed_started to sessions Entity

### PR DESCRIPTION
**Pull request by evanh**
The bucketed_started column is a virtual column, that ultimately gets
transformed into a function on the started column. However since it is a valid
column in the SnQL query, it should really be part of the Entity data model.

Also create a new test class that uses SnQL instead of legacy, that we can
start migrating tests into over time.

### CODE REVIEW

File: tests/test_sessions_api.py [hunk 1]
```diff
@@ -283,6 +283,136 @@ def test_minute_granularity_range(self, get_project_id: Callable[[], int]) -> No
         }
 
 
+class TestSessionsApi(BaseSessionsMockTest, BaseApiTest):
```
Comment (volokluev) [inline tests/test_sessions_api.py:286]: Are these tests related to the column change?

Comment (volokluev) [reply tests/test_sessions_api.py:286]: how would it fail if the column change wasn't made

Comment (evanh) [reply tests/test_sessions_api.py:286]: It won't on it's own. This is a small part of the larger initiative to start validating columns. I'm breaking that work down into more reviewable chunks. Ultimately I will turn on column validation for this entity, and at that point this test would start failing without the check.

Comment (volokluev) [reply tests/test_sessions_api.py:286]: So when validation is turned on this test is expected to fail? On what line? 

Also, is the generating of data necessary for the validation test?

Comment (evanh) [reply tests/test_sessions_api.py:286]: > So when validation is turned on this test is expected to fail? On what line?

This PR both introduces an error and fixes it. Turning the column validation to `WARN` would cause warning messages to get sent to Sentry, except that I added `bucketed_started` to the Entity data model, which should fix the warnings. So no, the test should pass and continue to pass. I added it because we weren't testing that code path at all.

> Also, is the generating of data necessary for the validation test?

I want to make sure that `bucketed_started` was returning bucketed data correctly.

**Review by volokluev — APPROVED** [review]
